File: src/shoppingcart/src/main/java/com/jll/repositories/ItemRepository.java
package com.jll.repositories;

import com.jll.model.Item;
import com.jll.model.ItemIdentity;
import com.jll.utilities.ConnectionManager;

public class ItemRepository extends Repository<Item, ItemIdentity> {
    public ItemRepository(ConnectionManager connectionManager) {
        super(connectionManager, Item.class, "item");
    }
}

File: src/ShoppingCartApi/DataAccess/ItemRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using ShoppingCartApi.Controllers.Item;
using ShoppingCartApi.Model;
using ShoppingCartEvents;

namespace ShoppingCartApi.DataAccess
{
    public class ItemRepository : Repository<Item, IItemEvent>, IItemRepository
    {
        public ItemRepository(string connectionString)
            : base(connectionString)
        {
        }

        protected override string TableName => "item";
        
        protected override Item MapEventsToEntity(Guid id, IReadOnlyList<IItemEvent> events) => new Item(id, events);
    }
}